src/index.ts

```diff
@@ -843,7 +843,7 @@ class ThreadPool {
         // When `isolateWorkers` is enabled, remove the worker after task is finished
         if (this.options.isolateWorkers && taskInfo.workerInfo) {
           this._removeWorker(taskInfo.workerInfo)
-          this._ensureMinimumWorkers()
+          this._ensureMaximumWorkers()
```

Comment:
This will solve the problem, but will create more workers than required.

```js
const pool = new Tinypool({
  filename: 'worker.js',
  minThreads: 0,
  maxThreads: 8,
  isolateWorkers: true,
});

await pool.run({});
```

* Pool is created with no workers.
* 1 worker created to run task.
* When task completes, `_ensureMaximumWorkers()` creates 8 new workers.
* None of these workers are ever given a task.
* They'll shut themselves down again shortly after starting (depending on `idleTimeout` option)

A more ideal solution would be that when a task completes:

* Check if there are any tasks in the queue *which don't already have a worker being started for them*.
* If not, do nothing.
* If there are, create 1 new worker.
* When that worker starts, it will run the queued task.

The first step is the tricky part. I don't think it's as simple as `this.taskQueue.length > 0`, because of circumstance where there are tasks in the queue, but they all already have a new worker being started for them.

I tried, but I can't figure out the logic of how a task is marked as "awaiting worker to start".